src/MembershipsSDKAdapter.js

```diff
@@ -38,6 +44,32 @@ export default class MembershipsSDKAdapter extends MembershipsAdapter {
     this.membershipSubjects = {};
   }
 
+  /**
+   * Gets members from a room
+   *
+   * @private
+   * @param {string} roomID ID of the room
+   * @returns {Array} List of members in a room
+   */
+  getRoomMembers(roomID) {
+    const roomMembers$ = new BehaviorSubject([]);
+
+    from(this.datasource.memberships.list({
```

Comment:
For promises we prefer to use the [`defer` operator](http://reactivex.io/documentation/operators/defer.html). This is because when you use `from` with a promise, the promise gets executed as soon as the code gets loaded and then the promise resolution gets cached, so the promise doesn't get executed again for the lifetime of the execution. When using `defer` the promise gets executed every time theres a new subscriber, as the promise resolution is deferred until it is called/subscribed.
In this case, if there happens to be an update the memberships of the room since the code started running, the observable using `from` won't pickup on it (since the promise resolved long back) and a new subscriber after the membership changed won't get the benefit of having this new information.

Reply:
There is that caching mechanism in place (`membershipSubjects`) that updates the member lists based on events and creates a single `BehaviorSubject` observable for each destination (room/meeting). So, in theory, a single backend call for loading the members would be needed, the rest being done via sdk events. However, because we are planning to reload the room members after each added/deleted event, it makes sense to be able to re-run this backend call. In any case, we'll keep this in mind and prefer defer over from.

Reply:
@cipak, you're correct, because we're caching we should technically only run the initial query once. Using `defer` has the side benefit of matching the mental execution flow when debugging. When we were first creating some of the other adapters and try to debug, we were really confused as to why the promise was running "off cycle", but then we realize why (what I explained above).

We should be careful to clean up the cache when the subject has no subscribers, as the subject would have completed at that point.